gradle/httpurlconnection-asynctask/app/src/main/java/com/irontec/kotlintest/MainActivity.kt

```diff
@@ -0,0 +1,91 @@
+package com.irontec.kotlintest
+
+import android.os.AsyncTask
+import android.support.v7.app.ActionBarActivity
+import android.os.Bundle
+import android.util.Log
+import android.view.Menu
+import android.view.MenuItem
+import android.widget.TextView
+import org.json.JSONObject
+import java.io.BufferedInputStream
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+
+
+public class MainActivity : ActionBarActivity() {
+
+    open val mTextView: TextView
+        get() {
+            return findViewById(R.id.text) as TextView
+        }
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        GetWeatherTask(mTextView).execute()
+    }
+
+    class GetWeatherTask(textView: TextView) : AsyncTask<Void, Void, String>() {
+
+        val innerTextView: TextView? = textView
+
+        override fun doInBackground(vararg params: Void?): String? {
+            val url = URL("https://raw.githubusercontent.com/irontec/android-kotlin-samples/master/common-data/bilbao.json")
+            val httpClient = url.openConnection() as HttpURLConnection
+            if(httpClient.getResponseCode() == HttpURLConnection.HTTP_OK){
```

Comment:
`if (httpClient.getResponseCode() == HttpURLConnection.HTTP_OK) {`
You can easily change all code formatting by using *Code* → *Reformat code*.